ScriptSpecification.md

```diff
@@ -122,66 +125,71 @@ Everything after the number will be ignored for compilation and follows no speci
-| 121 | Converts two literal values or variables to boolean and boolean and them.<br>```$local1 = true && true;```<br>```$local1 = $local2 && true;```<br>```$local1 = $local2 && $local3;``` |
-| 122 | Converts two literal values or variables to boolean and boolean or them.<br>```$local1 = true \|\| true;```<br>```$local1 = $local2 \|\| true;```<br>```$local1 = $local2 \|\| $local3;``` |
+| 120 | Coerces a literal value or variable to boolean and negates it.<br>```$local1 = not $local2;``` |
+| 121 | Coerces two literal values or variables to boolean and runs a logical AND operation on them.<br>```$local1 = true && true;```<br>```$local1 = $local2 && true;```<br>```$local1 = $local2 && $local3;``` |
+| 122 | Coerces two literal values or variables to boolean and runs a logical OR operation on them.<br>```$local1 = true \|\| true;```<br>```$local1 = $local2 \|\| true;```<br>```$local1 = $local2 \|\| $local3;``` |
 
 #### Strings
 | Type | Description |
 | - | - |
 | 500 | Was originally used to log a message in developement. Is a no-op in published games.<br>```$local1 = log("This is a message.");``` |
 | 501 | Formats a string with placeholder values.<br>```$local1 = format("Formatted message %s", $local2);``` |
-| 503 | Get a substring from another string.<br>```$local1 = substring("This message", 5);``` |
+| 502 | Converts a UTF-8 encoded string into a UTF-16 (wide) encoded string.<br>```$local1 = utf8_to_wide("hi");``` |
+| 503 | Get a substring from another string. Crashes on negative numbers. <br>```$local1 = substring("This message", 5);``` |
 
 #### Arrays
 | Type | Description |
 | - | - |
 | 530 | Creates a new multi-dimensional array.<br>```$local1 = new[2];```<br>```$local1 = new[2][1];``` |
-| 531 | Gets a reference to the indexed element in an array.<br>```$local1 = $local2[0];```<br>```$local1 = $local2[2];``` |
+| 531 | Gets a reference to the indexed element in an array. Returns `0` if a non-numeric type is indexed. <br>```$local1 = $local2[0];```<br>```$local1 = $local2[2];``` |
 
 The array index notation can be used in all shorthand assignments of type 240 - 271. They can not be directly used in operations of type 110 - 171. You need to use operation 531 to get an array element and set it to another variable to use them in those operations.
 
 #### Type
 | Type | Description |
 | - | - |
 | 40 | Gets the base type of a variable.<br>```$local1 = typeof($local2);``` |
-| 511 | Casts a literal value or variable to int.<br>```$local1 = (int)$local2;``` |
-| 512 | Casts a literal value or variable to bool.<br>```$local1 = (bool)$local2;``` |
-| 513 | Casts a literal value or variable to float.<br>```$local1 = (float)$local2;``` |
+| 511 | Casts a literal value or variable to int. Truncates floats; other types coerce to `0`. <br>```$local1 = (int)$local2;``` |
+| 512 | Casts a literal value or variable to bool. Non-Zero ints, floats, and arrays coerce to `true`. Strings coerce to `false`. <br>```$local1 = (bool)$local2;``` |
```

Comment:
To clarify for what coerces to false: "`0`, `0.0f`, and strings coerce to `false`."